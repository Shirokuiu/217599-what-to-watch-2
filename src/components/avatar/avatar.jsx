import React from "react";
import PropTupes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";

export const Avatar = (props) => {
  const {avatar} = props;

  return <div className="user-block">
    {avatar ?
      <Link to="/my-list" className="user-block__avatar">
        <img src={`https://htmlacademy-react-2.appspot.com${avatar}`} alt="User avatar" width="63" height="63"/>
      </Link> :
      <Link to="/login" className="user-block__link">Sign in</Link>
    }
  </div>;
};

Avatar.propTypes = {
  avatar: PropTupes.string
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  avatar: state.avatar
});

export default connect(mapStateToProps)(Avatar);
