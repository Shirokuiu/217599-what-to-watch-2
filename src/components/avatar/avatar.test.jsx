import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";

import {Avatar} from "../avatar/avatar";

it(`Is avatar rendered`, () => {
  const tree = renderer.create(<BrowserRouter>
    <Avatar
      avatar={`test`}
    />
  </BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
