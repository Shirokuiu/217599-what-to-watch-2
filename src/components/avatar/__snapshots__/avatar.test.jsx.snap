// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is avatar rendered 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__avatar"
    href="/my-list"
    onClick={[Function]}
  >
    <img
      alt="User avatar"
      height="63"
      src="https://htmlacademy-react-2.appspot.comtest"
      width="63"
    />
  </a>
</div>
`;
